data/modules/TradeShips/Trader.lua

```diff
@@ -275,6 +273,21 @@ Trader.removeFuel = function (ship, count)
 	return removed
 end
 
+Trader.checkDistBetweenStarport = function (ship)
+	local trader = Core.ships[ship]
+	if trader then
+		local distance
+		if trader.starport.type == "STARPORT_ORBITAL" then
+			distance = ship:DistanceTo(trader.starport)
+		else
+			local stationsParent = Space.GetBody(Core.ships[ship].starport.path:GetSystemBody().parent.index)
```

Comment:
might work `stationsParent = trader.starport.frameBody.body`

Reply:
`frameBody` only works for dynamic bodies. Station is not dynamic body, sadly.
`LuaBody.cpp`:
```
/*
 * Attribute: frameBody
 *
 * The non-dynamic body attached to the frame this dynamic body is in.
 *
 * Only valid for dynamic <Bodies>. For non-dynamic bodies <frameBody> will be
 * nil.
 ...
```

Reply:
Changed to this and also changed the way to get first system's star to:
`trader.starport:GetSystemBody().system:GetStars()[1].body`